GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using UnityEngine;
+
+namespace Turn_based_assets.Scripts
+{
+    public class MouseSelection : MonoBehaviour
+    {
+        private ISelection _selection;
+        [SerializeField] private GameObject cube;
+        public float distanceY;
+        public float offset;
+        public float gridSize;
+        private Plane _plane;
+        private Vector3 distanceFromCamera;
+        
+
+        private void Start()
+        {
+            var position = Camera.main.transform.position;
+            distanceFromCamera = new Vector3(position.x, position.y - distanceY,position.z);
+            _plane = new Plane(Vector3.up, distanceFromCamera);
+        }
+
+        private void Update()
+        {
+            Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+
+            float gridSpace = 0;
+
+            if (_plane.Raycast(ray, out gridSpace))
+            {
+                Vector3 gridPoint = ray.GetPoint(gridSpace);
+                gridPoint -= Vector3.one * offset;
+                gridPoint /= gridSize;
+                gridPoint = new Vector3(Mathf.Round(gridPoint.x), Mathf.Round(gridPoint.y), Mathf.Round(gridPoint.z));
+                gridPoint *= gridSize;
+                gridPoint += Vector3.one * offset;
+                if (cube != null) cube.transform.position = gridPoint;
+            }
+
+            if(Input.GetKeyDown(KeyCode.Mouse0))
```

Comment:
I'd probably move this entire if statement to a private method _or_ a local function so I could perform some `if` statement inversions.